sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -148,19 +151,53 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils with Adapti
         if (invalidBuckets.nonEmpty) {
           fail(s"Buckets ${invalidBuckets.mkString(",")} should have been pruned from:\n$plan")
         }
+
+        withSQLConf(SQLConf.BUCKETING_ENABLED.key -> "false") {
+          // Bucket pruning should still work when bucketing is disabled
```

Comment:
then do we have a config to turn on/off bucketing file pruning, or it's always applied?

Reply:
@cloud-fan - always applied. I can add one if this makes surprise to existing users/queries, and they can turn off if needed. e.g. `spark.sql.sources.bucketing.pruning.enabled`?

Reply:
Or we can keep the previous behavior: when `SQLConf.BUCKETING_ENABLED` is off, don't do any bucket optimization, including bucket scan and bucket pruning.

Reply:
@cloud-fan - make sense to me. Updated the change.